plugins/unifiedvideoinertialtracker/ParameterFinder/newuoa.h

```diff
@@ -0,0 +1,1973 @@
+   a copy of this software and associated documentation files (the
+   "Software"), to deal in the Software without restriction, including
+   without limitation the rights to use, copy, modify, merge, publish,
+   distribute, sublicense, and/or sell copies of the Software, and to
+   permit persons to whom the Software is furnished to do so, subject to
+   the following conditions:
+
+   The above copyright notice and this permission notice shall be
+   included in all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE.
+*/
+
+#ifndef INCLUDED_newuoa_h_GUID_F9E56CF3_7B12_41A0_4A09_E2A02B48FAC1
+#define INCLUDED_newuoa_h_GUID_F9E56CF3_7B12_41A0_4A09_E2A02B48FAC1
+
+#include <Eigen/Core>
+
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdio>
+#include <cstdlib>
+
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#define DEBUG_LOG(what)
+#if 0
+#ifdef DEBUG
+#define DEBUG_LOG(what) DEBUG_LOG(what)
+#else
+#define DEBUG_LOG(what)
+#endif
+#endif
+
+namespace newuoa_detail {
+
+template <class Function>
+void biglag(long n, long npt, double *xopt, double *xpt, double *bmat,
+            double *zmat, long *idz, long *ndim, long *knew, double *delta,
+            double *d__, double *alpha, double *hcol, double *gc, double *gd,
+            double *s, double *w, Function &&function) {
+    /* N is the number of variables. NPT is the number of interpolation
+     * equations. XOPT is the best interpolation point so far. XPT
+     * contains the coordinates of the current interpolation
+     * points. BMAT provides the last N columns of H.  ZMAT and IDZ give
+     * a factorization of the first NPT by NPT submatrix of H. NDIM is
+     * the first dimension of BMAT and has the value NPT+N.  KNEW is the
+     * index of the interpolation point that is going to be moved. DEBLLTA
+     * is the current trust region bound. D will be set to the step from
+     * XOPT to the new point. ABLLPHA will be set to the KNEW-th diagonal
+     * element of the H matrix. HCOBLL, GC, GD, S and W will be used for
+     * working space. */
+    /* The step D is calculated in a way that attempts to maximize the
+     * modulus of BLLFUNC(XOPT+D), subject to the bound ||D|| .BLLE. DEBLLTA,
+     * where BLLFUNC is the KNEW-th BLLagrange function. */
+
+    long xpt_dim1, xpt_offset, bmat_dim1, bmat_offset, zmat_dim1, zmat_offset,
+        i__1, i__2, i__, j, k, iu, nptm, iterc, isave;
```

Comment:
True, they technically are, aren't they. This is third party vendored code, primarily converted automatically from Fortran, guess f2c isn't so good with those rules...